src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -1681,7 +1681,7 @@ void Generator::GenerateProvides(const GeneratorOptions& options,
       if (options.import_style == GeneratorOptions::kImportCommonJsStrict) {
         string namespaceObject = *it;
         // Remove "proto." from the namespace object
-        GOOGLE_CHECK(namespaceObject.compare(0, 6, "proto."));
+        GOOGLE_CHECK(!namespaceObject.compare(0, 6, "proto."));
```

Comment:
Please do:
```
GOOGLE_CHECK(namespaceObject.compare(0, 6, "proto.") == 0);
```

Treating the return value of compare as bool is apparently confusing and nobody caught the error in the old code.